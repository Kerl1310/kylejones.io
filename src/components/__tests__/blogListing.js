import React from 'react';
import renderer from 'react-test-renderer';

import BlogListing from '../blogListing';

const testArticle = {
    node: {
        frontmatter: {
            date: "2021-02-04",
            path: "test-path",
            title: "Testing"
        },
        excerpt: "Testing excerpt"
    }
};

describe('Blog Listing', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<BlogListing article={testArticle} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
