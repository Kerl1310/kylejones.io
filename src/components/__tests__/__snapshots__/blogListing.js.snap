// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blog Listing renders correctly 1`] = `
<li
  className="blog-post"
  id="blog-post-2021-02-04"
>
  <div
    className="blog-post-date"
  >
    <h4>
      2/4/2021
    </h4>
  </div>
  <a
    href="/test-path"
    onClick={[Function]}
    onMouseEnter={[Function]}
    rel="noopener noreferrer"
    target="__blank"
  >
    Testing
  </a>
  <p>
    Testing excerpt
  </p>
  <hr
    className="sc-bdVaJa coSZVw"
  />
</li>
`;
